matchtree.go

```diff
@@ -888,11 +897,93 @@ func (d *indexData) newMatchTree(q query.Q) (matchTree, error) {
 			all:       regexp.String() == "(?i)(?-s:.)*",
 			matchTree: subMT,
 		}, nil
+
+	case *query.RepoBranches:
+		reposBranchesWant := make(map[uint16]uint64)
+		for _, r := range d.repoMetaData {
+			if branches, ok := s.Set[r.Name]; ok {
+				repoIdx, ok := d.repoMap[r.Name]
+				if !ok {
+					log.Panicf("repoMetaData not in sync with repoMap")
+				}
+				var mask uint64
+				for _, branch := range branches {
+					m, ok := d.branchIDs[repoIdx][branch]
+					if !ok {
+						continue
+					}
+					mask = mask | uint64(m)
+				}
+				if mask != 0 {
+					reposBranchesWant[repoIdx] = mask
+				}
```

Comment:
can you share code here with indexData.simplify?

Reply:
I replaced the map with a slice of masks. Since the zero value for mask (=0) has the meaning we can just remove the code you highlighted here.